src/lib/common/utils.test.ts

```diff
@@ -0,0 +1,16 @@
+import { test, expect } from 'vitest';
+import * as utils from './utils';
+
+test('checks correct discernTags output', () => {
+  expect(utils.discernTags([{ labels: [] }])).toEqual([]);
+  expect(utils.discernTags([{ labels: ['algo'] }])).toEqual(['algo']);
+  expect(
+    utils.discernTags([
+      { labels: [] },
+      { labels: ['george washington'] },
+      { labels: ['victor', 'delta', 'delta', 'delta'] },
+      { labels: ['kilo', 'victor'] },
+      { labels: ['delta', 'alpha', 'bravo'] },
+    ])
+  ).toEqual(['alpha', 'bravo', 'delta', 'george washington', 'kilo', 'victor']);
```

Comment:
1. The name of the unit test should describe the behavior we are checking for.
2. We should break up the unit tests to make them more readable.
3. We should use simpler sample data to test with (which makes it easier to understand at a glance).

Below is a unit test that tests the behavior expected on line 7 (testing the case where multiple labels are discerned in alphabetical order). The unit test below can be used as an example to make separate unit tests for the expectations on lines 5 (testing the case where the argument has no labels) and 6 (testing the case where 1 label is discerned).

```suggestion
test('discerns unique labels from list in alphabetical order', () => {
  const input = [
    { labels: [] },
    { labels: ['abc def'] },
    { labels: ['d', 'c', 'c', 'c'] },
    { labels: ['e', 'd'] },
    { labels: ['c', 'a', 'b'] },
  ];
  const expected = ['a', 'abc def', 'b', 'c', 'd', 'e'];
  const output = utils.discernTags(input);
  expect(output).toEqual(expected);
```

**Note**: Make sure you have 2 more unit test definitions representing the expectations on lines 5 and 6.

Reply:
The unit test has been updated.